netbench/netbench-collector/src/main.rs

```diff
@@ -18,24 +36,363 @@ pub struct Args {
 
     #[structopt(long, short, parse(try_from_str=parse_duration), default_value = "1s")]
     pub interval: Duration,
+
+    #[structopt(long, short)]
+    pub coordinate: bool,
+
+    #[structopt(long, required_if("coordinate", "true"))]
+    pub server_location: Option<String>,
+
+    #[structopt(long, required_if("coordinate", "true"))]
+    pub client_location: Option<String>,
+
+    #[structopt(long, required_if("coordinate", "true"))]
+    pub run_as: Option<String>,
+
+    #[structopt(long, short)]
+    pub verbose: bool,
 }
 
 impl Args {
     pub fn scenario(&self) -> Result<Scenario> {
         Scenario::open(std::path::Path::new(&self.scenario))
     }
+    pub fn as_server(&self) -> Option<bool> {
+        self.run_as.as_ref().map(|s| s.eq("server".into()))
+    }
+    pub fn as_client(&self) -> Option<bool> {
+        self.run_as.as_ref().map(|s| s.eq("client".into()))
+    }
+
+    pub fn location(&self) -> Option<String> {
+        if self.as_server()? {
+            self.server_location.clone()
+        } else if self.as_client()? {
+            self.client_location.clone()
+        } else {
+            unimplemented!("Only --run-as server and --run-as client are supported options");
```

Comment:
I think server/client are mutually exclusive options, at least for now, probably don't need the unimplemented bit.